propagation/http_trace_context_propagator_test.go

```diff
@@ -459,14 +459,14 @@ func TestInjectCorrelationContextToHTTPReq(t *testing.T) {
 			propagator.Inject(ctx, req.Header)
 
 			gotHeader := req.Header.Get("Correlation-Context")
-			wantedLen := len(strings.Join(tt.wantInHeader, ","))
-			if wantedLen != len(gotHeader) {
-				t.Errorf(
-					"%s: Inject Correlation-Context incorrect length %d != %d.", tt.name, tt.wantedLen, len(gotHeader),
-				)
+
+			headerParts := map[string]bool{}
```

Comment:
With these changes should we check if `headerParts` doesn't have anything extra? Like

```go
for _, inHeader := range tt.wantInHeader {
	if !headerParts[inHeader] { … }
	delete(headerParts, inHeader)
}
if len(headerParts) != 0 {
	t.Errorf("too much leftover stuff in headers: %#v", headerParts)
}
```